src/logic/assetRetriever.ts

```diff
@@ -0,0 +1,13 @@
+import { createFetchComponent } from '../adapters/fetch'
+
+export default async function createAssetRetriever(
+  contentHost: string
+): Promise<(hash: string) => Promise<ArrayBuffer>> {
+  const fetcher = await createFetchComponent()
```

Comment:
Fetcher should be an argument along with config

```suggestion
  const fetcher = components.fetch
  const contentHost = (await components.config.getString('CONTENT_URL')) || ''
```